File: app/src/main/java/com/buddha/mindboard/module/details/DetailsModule.java
package com.buddha.mindboard.module.details;

import com.buddha.mindboard.di.scope.FragmentScoped;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class DetailsModule {
    @FragmentScoped
    @ContributesAndroidInjector
    public abstract DetailsFragment detailsFragment();
}

File: app/src/main/java/com/buddha/mindboard/module/details/DetailsModule.kt
package com.buddha.mindboard.module.details

import com.buddha.mindboard.di.scope.FragmentScoped
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class DetailsModule {
    @FragmentScoped
    @ContributesAndroidInjector
    abstract fun detailsFragment(): DetailsFragment
}
